twisted/pydoctor
# (Fixes #331) Add project version to Sphinx object inventory.

Fixes #331

This updated the sphinx inventory implementation to support project version.

Previously it was hardcoded to version 2.0


Changes
=======

* Add `--project-version` to specify a fixed version,
* Refactor `parse_args` as a high level function for handling the CLI options and preparing them for internal usage.

As a drive by, I have updated the tox default env list and Contributing docs, so that you can just run `tox -p all` and have all the "recommended" env executed before pushing your changes.


We tried extracting the info from project meta file, but decided to move to a separate PR


How to test
=========

`tox -e apidocs` builds the version and name from metadata.

`strings build/apidocs/objects.inv | head` to check the version in Sphinx object inventory.

`tox -e docs` to build with version and name from both metadata and custom CLI args (for epydoc sample).

On read the docs. go here https://pydoctor--332.org.readthedocs.build/en/332/usage.html#building-pydoctor-together-with-sphinx-html-build

And then go with the mouse over : This can be done by using the **pydoctor.sphinx_ext.build_apidocs** extension.

You should see the current pydoctor version.
